04.ls/ls.rb

```diff
@@ -0,0 +1,30 @@
+#! /usr/bin/env ruby
+# frozen_string_literal: true
+
+COLUMNS = 3
+COLUMNS_DISTANCE = 5
+
+def output(make_output_file_names, file_names)
+  longest_string = file_names.max_by(&:size)
+  string_width = longest_string.size + COLUMNS_DISTANCE
+  make_output_file_names.each do |row|
+    row.each do |file_name|
+      print file_name.to_s.ljust(string_width)
+    end
+    puts
+  end
+end
+
+def make_output_file_names
+  rows = file_names.size.ceildiv(COLUMNS)
+  additional_blanks = COLUMNS * rows - file_names.size
+  full_file_names = file_names + Array.new(additional_blanks)
+  sliced_file_names = full_file_names.each_slice(rows).to_a
+  sliced_file_names.transpose
+end
+
+def file_names
+  Dir['*'].sort_by(&:downcase)
+end
+
+output(make_output_file_names, file_names)
```

Comment:
ここを起点にプログラムが始まるのであれば、main/run/execみたいな名前のメソッドがいいと思います。
https://qiita.com/jnchito/items/4b4cae54170cc2f4377e

また、ここでは引数に応じて柔軟性を持たせる必要がないので、引数はなくしてしまっていいと思います。
(mainメソッド内で呼び出す)

```suggestion
main
```

Reply:
プログラムの起点となるメソッドをmainメソッドとして定義しました。
引数を削除しました。